neuro_san/service/http/handlers/base_request_handler.py

```diff
@@ -97,6 +99,26 @@ def get_metadata(self) -> Dict[str, Any]:
                 result[item_name] = "None"
         return result
 
+    async def get_service(self, agent_name: str, metadata: Dict[str, Any]) -> AsyncAgentService:
+        """
+        Get agent's AsyncAgentService for request execution
+        :param agent_name: agent name
+        :param metadata: metadata to be used for logging if necessary.
+        :return: instance of AsyncAgentService if it is defined for this agent
+                 None otherwise
+        """
+        update_done: bool = await self.update_agents(metadata=metadata)
+        if not update_done:
+            return None
+
+        service_provider: AsyncAgentServiceProvider = self.agent_policy.allow(agent_name)
+        if service_provider is None:
+            self.set_status(404)
+            self.logger.error(metadata, "error: Invalid request path %s", self.request.path)
+            self.do_finish()
+            return None
+        return service_provider.get_service()
+
```

Comment:
Factor out common logic for getting AsyncAgentService for request processing.
Will be used in all (3) neuro-san API requests.